### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by sam2if**
### Project requirements
- Add a column `species` of type `string` to your `animals` table. Modify your `schema.sql` file.
- Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Remember to add these insertions to your `data.sql` file.

----

- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.
- **Take a screenshot of the results of your actions.**
- Inside a transaction:
  - Update the `animals` table by setting the `species` column to `digimon` for all animals that have a name ending in `mon`.
  - Update the `animals` table by setting the `species` column to `pokemon` for all animals that don't have `species` already set.
  - Verify that changes were made.
  - Commit the transaction.
  - Verify that changes persist after commit.
- **Take a screenshot of the results of your actions.**
- Now, take a deep breath and... **Inside a transaction** delete all records in the `animals` table, then roll back the transaction.
- After the rollback verify if all records in the `animals` table still exists. After that, you can start breathing as usual ;) 
- **Take a screenshot of the results of your actions.**
- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- **Take a screenshot of the results of your actions.**
- Remember to add all these transactions to your `queries.sql` file.

----
- Write queries to answer the following questions:
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- **Take a screenshot of the results of your queries.**
- Remember to add these queries to your `queries.sql` file.
- Remember to add screenshots with the results of your queries and with the results of your play with transactions to your Pull Request description.

### THE FOLLOWING ARE THE SCREENSHOTS OF THE RESULTS FROM QUERIES.

Screenshot (74)
Screenshot (75)
Screenshot (76)
Screenshot (77)
Screenshot (78)
Screenshot (79)
Screenshot (80)
Screenshot (81)
Screenshot (82)
Screenshot (83)
Screenshot (84)
Screenshot (85)
Screenshot (86)
Screenshot (87)
Screenshot (88)
Screenshot (89)
Screenshot (90)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -5,4 +5,27 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-SELECT * FROM animals WHERE (weight_kg >= 10.4 AND weight_kg <= 17.3);
\ No newline at end of file
+SELECT * FROM animals WHERE (weight_kg >= 10.4 AND weight_kg <= 17.3);
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
Comment (danmainah) [inline queries.sql]: - [ ] @sam2if Great work adding a query to add data on species column, however, you forgot to add a query `SELECT species from animals; ` before and after ROLLBACK  to verify the changes were made.
image

File: queries.sql [hunk 2]
```diff
@@ -5,4 +5,27 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
```
Comment (danmainah) [inline queries.sql]: - [ ] @sam2if Great work adding a query to add data on species column, however, you forgot to add a query `SELECT species from animals; ` before and after COMMIT to verify the changes were made.

File: queries.sql [hunk 3]
```diff
@@ -5,4 +5,27 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (danmainah) [inline queries.sql]: - [ ] @sam2if Great work making changes in your database, however, you forgot to add a query `SELECT COUNT(*) FROM ANIMALS;`  before and after ROLLBACK to verify the changes were made.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @sam2if , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added the data given correctly to the database
✔️ Add command to add column species of type string to animals table correctly.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @sam2if 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All queries are now present :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**